app/controllers/groups_controller.rb

```diff
@@ -18,6 +18,31 @@ def show
     end
   end
 
+  def group_token_check
+    @group = Group.where("id=?", params[:id])
+    @group.group_token_generate
+    render json: { "updated" => @group }
+  end
+
+  def group_invite
+    @group = Group.find(params[:id])
+    @current_user = current_user.id
+    if Group.where("group_token = ? AND updated_at >= ?", params[:group_token], 12.days.ago).exists?
```

Comment:
Add a rails scope with name with_valid_token

```suggestion
    if Group.with_valid_token.where(token: params[:token]).exists?
```

Reply:
add scope with_valid_token